src/key.rs

```diff
@@ -85,6 +94,27 @@ impl PublicKey {
     pub fn address(&self) -> &[u8; 20] {
         &self.address
     }
+
+    /// Checks if given `signature` is a valid ECDSA signature for `message` using the this public key.
+    /// Returns `Ok(true)` on success.
+    pub fn verify(self, signature: &Signature, message: &[u8]) -> Result<bool, secp256k1::Error> {
```

Comment:
`&self` would do, I don't think we need to consume it here

Reply:
ok